### ISSUE TITLE
[Tap To Pay] UNEXPECTED_ERROR.UNEXPECTED_SDK_ERROR - KEY_ERROR (null): Error setting key entry during secure import

### ISSUE THREAD

**Issue by marcoripa96** [stripe/stripe-terminal-react-native#773]
**Describe the bug**
When discovering readers (`discoverReaders`) with `simulated` set to `false` on some Android devices:

```ts
const { error } = await discoverReaders({
    discoveryMethod: "localMobile",
    simulated: getTapSimulation(),
});
```
I get the following error:

```js
{ 
   code: 'UNEXPECTED_ERROR.UNEXPECTED_SDK_ERROR',
   message: 'KEY_ERROR (null): Error setting key entry during secure import' 
}

```
**To Reproduce**

1. Call `discoverReaders` with `simulated` set to `false`

**Expected behavior**
The `discoverReaders` function should discover new readers without giving that error

**Stripe Terminal React Native SDK version**

- `0.0.1-beta.20`

**Smartphone (please complete the following information):**

I only tested the applicattion on Android devices and here's the list that raise the error:

- Device: OPPO A94 5G
- Version: 13

- Device: OPPO A 94 5g
- Version 13

- Device: REAL ME 8
- Version: 12

- Device: VIVO Y72
- Version: 11

**Comment by nazli-stripe** [comment]
This can happen if the device the app is running on has some aggressive memory management. Looks like all devices in this instance kill background services in some situations. I recommend taking a look at [dontkillmyapp.com](https://dontkillmyapp.com/) for your devices and following the recommendations. We'll update the error message and documentation for this case, and further look into improving `supportsReadersOfType` method to run secure keystore imports check.

**Comment by marcoripa96** [comment]
Sorry if I comment on the closed issue, but isn't this related to the following, which should be solved using the sdk version 3.8.0?

> Fix: Prevent a crash when attempting to connect to a mobile reader on Android devices that do not support Android Keystore cryptographic operations. Fixes https://github.com/stripe/stripe-terminal-android/issues/466.

**Comment by nazli-stripe** [comment]
the issue there is an `OfflineDecryptionException` exception, which seems to be different from what you are reporting. beta.21 release of RN SDK will be on 3.8.0 in case that fixes things.
